x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/EnterpriseSearchUsageTransportAction.java

```diff
@@ -0,0 +1,109 @@
+import org.elasticsearch.xpack.core.application.EnterpriseSearchFeatureSetUsage;
+
+import java.util.Collections;
+import java.util.Map;
+
+public class EnterpriseSearchUsageTransportAction extends XPackUsageFeatureTransportAction {
+    private static final Logger logger = LogManager.getLogger(EnterpriseSearchUsageTransportAction.class);
+    private final XPackLicenseState licenseState;
+    private final Client client;
+
+    private final boolean enabled;
+
+    @Inject
+    public EnterpriseSearchUsageTransportAction(
+        TransportService transportService,
+        ClusterService clusterService,
+        ThreadPool threadPool,
+        ActionFilters actionFilters,
+        IndexNameExpressionResolver indexNameExpressionResolver,
+        Settings settings,
+        XPackLicenseState licenseState,
+        Client client
+    ) {
+        super(
+            XPackUsageFeatureAction.ENTERPRISE_SEARCH.name(),
+            transportService,
+            clusterService,
+            threadPool,
+            actionFilters,
+            indexNameExpressionResolver
+        );
+        this.licenseState = licenseState;
+        this.client = client;
+        this.enabled = XPackSettings.ENTERPRISE_SEARCH_ENABLED.get(settings);
+    }
+
+    @Override
+    protected void masterOperation(
+        Task task,
+        XPackUsageRequest request,
+        ClusterState state,
+        ActionListener<XPackUsageFeatureResponse> listener
+    ) {
+        if (enabled == false) {
+            EnterpriseSearchFeatureSetUsage usage = new EnterpriseSearchFeatureSetUsage(
+                LicenseUtils.LICENSED_ENT_SEARCH_FEATURE.checkWithoutTracking(licenseState),
+                enabled,
+                Collections.emptyMap()
+            );
+            listener.onResponse(new XPackUsageFeatureResponse(usage));
+            return;
+        }
+
+        try {
+            ListSearchApplicationAction.Response resp = client.execute(
+                ListSearchApplicationAction.INSTANCE,
+                new ListSearchApplicationAction.Request(null, new PageParams(0, 0))
+            ).get();
+            listener.onResponse(
+                new XPackUsageFeatureResponse(
+                    new EnterpriseSearchFeatureSetUsage(
+                        enabled,
+                        LicenseUtils.LICENSED_ENT_SEARCH_FEATURE.checkWithoutTracking(licenseState),
+                        Map.of("count", resp.queryPage().count())
+                    )
+                )
+            );
+        } catch (Exception e) {
+            logger.warn("Failed to get search application count to include in Enterprise Search usage", e);
+            EnterpriseSearchFeatureSetUsage usage = new EnterpriseSearchFeatureSetUsage(
+                LicenseUtils.LICENSED_ENT_SEARCH_FEATURE.checkWithoutTracking(licenseState),
+                enabled,
+                Collections.emptyMap()
+            );
+            listener.onResponse(new XPackUsageFeatureResponse(usage));
```

Comment:
Silently ignoring exceptions like this seems wrong. I [see](https://github.com/elastic/elasticsearch/blob/main/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/action/TransportXPackInfoAction.java#L83) that simply throwing the exception could break the entire info action call so that's not an option either. 
It's also a bigger scope than just adding a new info endpoint for ent-search so it's fine to keep it this way in this PR. 
I just wanted to raise the awareness that this is not something we should do in general ;) 
I'll open a more generic issue to discuss error handling in the `XPackUsageFeatureTransportAction`.

Reply:
for what it's worth this is a pattern used elsewhere: https://github.com/elastic/elasticsearch/blob/7fb344238dcad615282b70557b70c67eede482ca/x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/MachineLearningUsageTransportAction.java#L131-L150